src/main/scala/chisel3/internal/firrtl/Emitter.scala

```diff
@@ -82,7 +82,7 @@ private class Emitter(circuit: Circuit) {
     */
   private def emit(m: Component): String = {
     // Generate the body.
-    val moduleName = m.id.getClass.getName.split('.').last
+    val moduleName = m.id.desiredName
```

Comment:
Correct me if I'm wrong, but if name is the legalized form of desiredName, shouldn't that be what we use here?